shallow_backup/backup.py

```diff
@@ -172,19 +173,17 @@ def run_cmd_if_no_dry_run(command, dest, dry_run) -> int:
 
 	# npm
 	print_pkg_mgr_backup("npm")
-	command = "npm ls --global --parseable=true --depth=0"
-	temp_file_path = f"{backup_path}/npm_temp_list.txt"
+	command = "npm ls --global --json=true --depth=0"
+	temp_file_path = f"{backup_path}/npm_temp_list.json"
```

Comment:
1. The filename change would need to be made in `reinstall.py`
2. The old reinstallation process would throw an error without some changes

Let's walk through it:

https://github.com/alichtman/shallow-backup/blob/b570f9d176a51878a97033915cab6b8216f48a12/shallow_backup/reinstall.py#L142-L145

Here's some example output from `$ npm ls --global --json=true --depth=0 > npm-json.txt`


```json
{
  "name": "lib",
  "dependencies": {
    "corepack": {
      "version": "0.15.3",
      "overridden": false
    },
    "insect": {
      "version": "5.8.0",
      "overridden": false
    },
    "md-to-pdf": {
      "version": "5.2.1",
      "overridden": false
    },
    "npm": {
      "version": "9.3.1",
      "overridden": false
    }
  }
}
```

```bash
$ cat npm-json.txt | xargs npm install -g
npm ERR! code EINVALIDTAGNAME
npm ERR! Invalid tag name "{" of package "{": Tags may not have any characters that encodeURIComponent encodes.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/alichtman/.cache/npm/_logs/2023-03-06T00_26_49_097Z-debug-0.log
```

In order for this change to work, we'd need to parse out the modules *from* the `json` file we're creating (which is roughly what the logic that was removed from `backup.py` was doing).